Analysis/interface/AnalysisTypes.h

```diff
@@ -62,8 +62,12 @@ ENUM_NAMES(SampleType) = {
     { SampleType::Data, "Data" }, { SampleType::MC, "MC" }, { SampleType::Embedded, "Embedded" }
 };
 
-inline TauType GenMatchToTauType(GenLeptonMatch gen_match, SampleType sample_type)
+inline TauType GenMatchToTauType(GenLeptonMatch gen_match, SampleType sample_type, const int genLepton_index, const int genJet_index)
```

Comment:
I suggest keeping the `GenMatchToTauType` definition as it was. And return `boost::optional<GenLeptonMatch>` in `PassGenLeptonCut`